codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -69,60 +69,113 @@
     private final Http2Settings initialSettings;
     private ChannelFutureListener closeListener;
     private BaseDecoder byteDecoder;
-    private long gracefulShutdownTimeoutMillis = DEFAULT_GRACEFUL_SHUTDOWN_TIMEOUT_MILLIS;
+    private long gracefulShutdownTimeoutMillis;
 
-    public Http2ConnectionHandler(boolean server) {
-        this(server, true);
+    public static final class Builder extends BuilderBase<Http2ConnectionHandler, Builder> {
+        @Override
+        public Http2ConnectionHandler build(Http2ConnectionDecoder decoder,
+                                            Http2ConnectionEncoder encoder) {
+            Http2ConnectionHandler handler = new Http2ConnectionHandler(decoder, encoder, initialSettings());
+            setPostBuildOptions(handler);
+            return handler;
+        }
     }
 
-    public Http2ConnectionHandler(boolean server, boolean validateHeaders) {
-        this(new DefaultHttp2Connection(server), validateHeaders);
-    }
+    public abstract static class BuilderBase<T extends Http2ConnectionHandler, B extends BuilderBase<T, B>> {
+        private Http2Settings initialSettings;
```

Comment:
Your suggestion in https://github.com/netty/netty/pull/4299#discussion_r40876127 may make more sense now. You are suggesting disallowing the `null` initial settings and provide some default one?